aten/src/ATen/Parallel.h

```diff
@@ -37,56 +26,17 @@ CAFFE2_API int get_num_threads();
 
 // Returns the current thread number (starting from 0)
 // in the current parallel region, or 0 in the sequential region
-inline int get_thread_num() {
-#ifdef _OPENMP
-  return omp_get_thread_num();
-#else
-  return 0;
-#endif
-}
+CAFFE2_API int get_thread_num();
 
-inline bool in_parallel_region() {
-#ifdef _OPENMP
-  return omp_in_parallel();
-#else
-  return false;
-#endif
-}
+// Checks whether the code runs in parallel region
+CAFFE2_API bool in_parallel_region();
```

Comment:
Is "CAFFE2_API" the right thing to use in aten?

Reply:
I guess it's used everywhere else...

Reply:
It is, as of the moment ATen is compiled into libcaffe2